clippy_lints/src/shadow.rs

```diff
@@ -291,14 +291,15 @@ fn lint_shadow<'tcx>(
                 },
             );
         } else {
+            let truncated_span_lines_num = 5;
             span_lint_and_then(
                 cx,
                 SHADOW_UNRELATED,
                 pattern_span,
                 &format!(
                     "`{}` is shadowed by `{}`",
                     snippet(cx, pattern_span, "_"),
-                    snippet(cx, expr.span, "..")
+                    snippet_truncated(cx, expr.span, "..", truncated_span_lines_num)
```

Comment:
TL;DR: Just remove the second snippet:

```suggestion
                    "`{}` is shadowed",
                    snippet(cx, pattern_span, "_"),
```